dashboard/state_aggregator.py

```diff
@@ -195,6 +197,35 @@ async def list_objects(self, *, option: ListApiOptions) -> dict:
         result.sort(key=lambda entry: entry["object_id"])
         return {d["object_id"]: d for d in islice(result, option.limit)}
 
+    async def list_runtime_envs(self, *, option: ListApiOptions) -> List[dict]:
+        """List all runtime env information from the cluster.
+
+        Returns:
+            A list of runtime env information in the cluster.
+            We don't have id -> data mapping like other API because runtime env
+            doesn't have unique ids.
+        """
+        replies = await asyncio.gather(
+            *[
+                self._client.get_runtime_envs_info(node_id, timeout=option.timeout)
+                for node_id in self._client.get_all_registered_agent_ids()
+            ]
+        )
+        result = []
+        for reply in replies:
+            states = reply.runtime_env_states
+            for state in states:
+                data = self._message_to_dict(message=state, fields_to_decode=[])
+                data["runtime_env"] = RuntimeEnv.deserialize(
+                    data["runtime_env"]
+                ).to_dict()
+                data = filter_fields(data, RuntimeEnvState)
+                result.append(data)
+
+        # Sort to make the output deterministic.
+        result.sort(key=lambda entry: entry["ref_cnt"])
```

Comment:
why do we sort on ref_cnt? Seems it's arbitrary. I guess if we want to sort, user should provide the sort field?

Reply:
or sort on creation time, that seems like the most intuitive.

Reply:
It is arbitrary. Creation time is not always available (if the runtime env is not created). But now that I think it's possible to sort by creation time, and tasks without creation time (failed or not created yet) as a highest priority since that will be runtime env consumers would care the most for debugging purpose.

 I think we need more discussion on if we'd like to allow users to sort because with the current proposal (limiting the output size for query), we cannot do full global sort (e.g., to sort tasks or runtime env correctly, we need to be distributed sort). I am not against just providing this as a flag, but we need some API consensus here.

Reply:
Hmm maybe two better options than ref count; 

1. Add id to runtime env, and just sort by id like other APIs
2. By creation time as @edoakes, and set entries without creation time (meaning it is not created yet, or failed to be created) as the highest priority.

Reply:
this could be changed to create_time now?

Reply:
Yeah this sgtm. I will do based on this policy;

no creation time -> highest
sort by higher creation time order (since longer creation => more likely there was an issue )